Trappist/src/Promact.Trappist.Web/wwwroot/app/tests/tests-dashboard/test-create-dialog.component.ts

```diff
@@ -1,8 +1,55 @@
 ﻿import { Component } from '@angular/core';
+import { MdDialog, MdDialogRef, MdSnackBar } from '@angular/material';
+import { Test } from '../tests.model';
+import { TestService } from '../tests.service';
+import { Response } from '../tests.model';
+import { TestsDashboardComponent } from './tests-dashboard.component';
 
 @Component({
     moduleId: module.id,
     selector: 'test-create-dialog',
     templateUrl: 'test-create-dialog.html'
 })
-export class TestCreateDialogComponent { }
+
+export class TestCreateDialogComponent {
+    responseObj: boolean;
+    errorMessage: boolean;
+    testNameReference: string;
+    test: Test;
+    errorStatus: any;
+    constructor(public dialogRef: MdDialogRef<TestCreateDialogComponent>, private testService: TestService, private snackbar: MdSnackBar ) {
+        this.test = new Test();
+    }
+    /**
+     * this method is used to add a new test
+     * @param testNameRef is name of the test
+     */
+    AddTest(testNameRef: string) {
+        this.test.testName = testNameRef;
```

Comment:
what is the need of assignment? it can be passed directly in the method. if any other purpose is there, then let me know.